src/PowerShellEditorServices/Utility/PSCommandExtensions.cs

```diff
@@ -55,10 +55,13 @@ public static PSCommand AddDebugOutputCommand(this PSCommand psCommand)
 
         public static PSCommand MergePipelineResults(this PSCommand psCommand)
         {
-            // We need to do merge errors and output before rendering with an Out- cmdlet
-            Command lastCommand = psCommand.Commands[psCommand.Commands.Count - 1];
-            lastCommand.MergeMyResults(PipelineResultTypes.Error, PipelineResultTypes.Output);
-            lastCommand.MergeMyResults(PipelineResultTypes.Information, PipelineResultTypes.Output);
+            if (psCommand.Commands.Count > 0)
+            {
+                // We need to do merge errors and output before rendering with an Out- cmdlet
+                Command lastCommand = psCommand.Commands[psCommand.Commands.Count - 1];
+                lastCommand.MergeMyResults(PipelineResultTypes.Error, PipelineResultTypes.Output);
+                lastCommand.MergeMyResults(PipelineResultTypes.Information, PipelineResultTypes.Output);
+            }
```

Comment:
I'm kinda curious what circumstance we're calling this method without any commands added though 🤔

Reply:
With `throw`, there was no "previous command output" and that's where the null was coming from.

Reply:
https://github.com/JustinGrote/PowerShellEditorServices/blob/ceb20e8f47754f5c829c6f4ae9c102241f7e59da/src/PowerShellEditorServices/Services/PowerShell/Execution/SynchronousPowerShellTask.cs#L106-L109